zerver/lib/actions.py

```diff
@@ -2044,6 +2055,29 @@ def get_default_subs(user_profile):
     # to some day further customize how we set up default streams for new users.
     return get_default_streams_for_realm(user_profile.realm)
 
+def make_dict(row):
+    # type: (Stream) -> Dict[str, Any]
+        return dict(
+        stream_id = row.id,
+        name = row.name,
+        description = row.description,
+        invite_only = row.invite_only,
+        )
+
+# returns default streams in json serializeable format
+def json_serialize_streams(streams):
+    # type: (List[Stream]) -> List[Dict[str, Any]]
+
+    json_serialized_streams = []
+
+    for stream in streams:
+        temp_stream = make_dict(stream)
+        json_serialized_streams.append(temp_stream)
+
+    json_serialized_streams.sort(key=lambda elt: elt["name"])
+
+    return json_serialized_streams
```

Comment:
I think it shouldn't have `json` in the name since it doesn't use JSON at all -- would call this something more like `streams_to_dicts`.  

Actually, reading this more closely, I think your `make_dict(stream)` function already exists as stream.to_dict(), so probably you can get rid of that entirely.  So here's a proposal:

* Do a commit at the start of this branch that replaces `make_dict` below in do_get_streams() with `stream.to_dict()` -- that's just a code cleanup.
* replace this function with a simple function `streams_to_dicts_sorted` that I think you can write as just this:
`sorted([stream.to_dict() for stream in streams], key=lambda elt: elt["name"])`, and use that in the two places below.